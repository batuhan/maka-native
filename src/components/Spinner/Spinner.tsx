import React, { ReactNode } from 'react'
import { View, Text } from 'react-native'

interface Props {
  children?: ReactNode
}

const Spinner = ({ children }: Props) => {
  return <View>{children || <Text>Loading...</Text>}</View>
}

export default Spinner
