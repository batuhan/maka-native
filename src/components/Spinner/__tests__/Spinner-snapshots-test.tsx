import React from 'react'
import renderer from 'react-test-renderer'
import { Text } from 'react-native'
import Spinner from '../Spinner'

describe('Spinner', () => {
  it('renders correctly', () => {
    const instance = renderer.create(
      <Spinner>
        <Text>Loading...</Text>
      </Spinner>,
    )

    expect(instance.toJSON()).toMatchSnapshot()
  })
})
