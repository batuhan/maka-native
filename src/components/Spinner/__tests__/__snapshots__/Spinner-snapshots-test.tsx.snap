// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders correctly 1`] = `
<View>
  <Text>
    Loading...
  </Text>
</View>
`;
